ZAPD/ZDisplayList.cpp

```diff
@@ -2033,15 +2026,194 @@ string ZDisplayList::GetSourceOutputCode(const std::string& prefix)
+			parent->defines += defines;
+
+		// Generate Vertex Declarations
+		vector<int32_t> vtxKeys;
+
+		for (pair<int32_t, vector<Vertex>> item : vertices)
+			vtxKeys.push_back(item.first);
+
+		// for (pair<int32_t, vector<Vertex>> item : vertices)
+		for (int i = 0; i < vtxKeys.size(); i++)
+		{
+			vector<Vertex> item = vertices[vtxKeys[i]];
+
+			// if (i < vtxKeys.size() - 1)
+			//{
+			//	vector<Vertex> nextItem = vertices[vtxKeys[i + 1]];
+
+			//	int endAddr = (vtxKeys[i] + (item.size() * 16));
+			//	if (endAddr != vtxKeys[i + 1] && parent->GetDeclarationRanged(endAddr) == nullptr)
+			//	{
+			//		for (Vertex vtx : nextItem)
+			//		{
+			//			item.push_back(vtx);
+			//		}
+
+			//		vtxKeys.erase(vtxKeys.begin() + (i + 1));
+			//		vertices.erase(vertices.begin() + (i + 1));
+			//		//i--;
+			//	}
+			//}
+
+			string declaration = "";
+
+			int curAddr = vtxKeys[i];
+
+			for (Vertex vtx : item)
+			{
+				if (curAddr != vtxKeys[i])
+					declaration += "\n";
+
+				declaration +=
+					StringHelper::Sprintf("    VTX(%i, %i, %i, %i, %i, %i, %i, %i, %i),", vtx.x,
+				                          vtx.y, vtx.z, vtx.s, vtx.t, vtx.r, vtx.g, vtx.b, vtx.a);
+
+				curAddr += 16;
+			}
+
+			vtxDeclarations[vtxKeys[i]] = declaration;
+
+			if (parent != nullptr)
+			{
+				std::string vtxName =
+					StringHelper::Sprintf("%sVtx_%06X", prefix.c_str(), vtxKeys[i]);
+				std::string incStr = StringHelper::Sprintf(
+					"%s/%s.%s.inc", Globals::Instance->outputPath.c_str(), vtxName.c_str(), "vtx");
+				parent->AddDeclarationArray(vtxKeys[i], DeclarationAlignment::None,
+				                            item.size() * 16, "static Vtx", vtxName, item.size(),
+				                            declaration);
+
+				Declaration* vtxDecl = parent->AddDeclarationIncludeArray(
+					vtxKeys[i], incStr, item.size() * 16, "static Vtx", vtxName, item.size());
+				vtxDecl->isExternal = true;
+			}
+		}
+	}
+
+	if (parent != nullptr)
 		return "";
+
+	return sourceOutput;
+}
+
+std::string ZDisplayList::ProcessLegacy(const std::string& prefix)
+{
+	char line[4096];
+	string sourceOutput = "";
+
+	for (int i = 0; i < instructions.size(); i++)
+	{
+		uint8_t opcode = (uint8_t)(instructions[i] >> 56);
+		uint64_t data = instructions[i];
+		sourceOutput += "    ";
+
+		auto start = chrono::steady_clock::now();
+
+		int optimizationResult = OptimizationChecks(i, sourceOutput, prefix);
+
+		if (optimizationResult != -1)
+		{
+			i += optimizationResult - 1;
+			line[0] = '\0';
+		}
+		else
+		{
+			if (dListType == DListType::F3DZEX)
+				ParseF3DZEX((F3DZEXOpcode)opcode, data, i, prefix, line);
+			else
+				ParseF3DEX((F3DEXOpcode)opcode, data, i, prefix, line);
+		}
+
+		auto end = chrono::steady_clock::now();
+		auto diff = chrono::duration_cast<chrono::milliseconds>(end - start).count();
+
+		if (Globals::Instance->verbosity >= VERBOSITY_DEBUG && diff > 5)
+			printf("F3DOP: 0x%02X, TIME: %ims\n", opcode, diff);
```

Comment:
`diff` is a `int64_t`, so it should be printed with `%li` instead of `%i`
```suggestion
			printf("F3DOP: 0x%02X, TIME: %lims\n", opcode, diff);
```